contracts/orbital-core/src/utils.rs

```diff
@@ -0,0 +1,83 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, Coin, MessageInfo, StdError, StdResult, Uint128, Uint64};
+use cw_utils::must_pay;
+use neutron_sdk::bindings::msg::IbcFee;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::error::ContractError;
+
+pub fn assert_fee_payment(info: &MessageInfo, expected_fee: &Coin) -> Result<(), ContractError> {
+    match must_pay(info, &expected_fee.denom) {
+        Ok(amt) => ensure!(
+            amt >= expected_fee.amount,
+            ContractError::DomainRegistrationError("insufficient fee".to_string())
+        ),
+        Err(_) => {
+            return Err(ContractError::DomainRegistrationError(format!(
+                "no funds sent; expected {}.",
+                expected_fee
+            )))
+        }
+    };
+    Ok(())
+}
```

Comment:
who custom function instead of `must_pay`?

Reply:
there is `must_pay` under the hood, but it also needs to validate that the paid amount was at least as much as we queried from chain (ica registration fee). updated it to be cleaner now though.

Reply:
Yea, sorry missed the `must_pay`